Hamilton_Matthew/index.py

```diff
@@ -0,0 +1,91 @@
+from django.conf import settings
+from django_mako_plus import view_function, jscontext
+from datetime import datetime, timezone
+from django.http import HttpResponseRedirect
+from django.shortcuts import render
+from catalog import models as cmod
+
+@view_function
+def process_request(request, category_id):
+
+    if not request.user.is_authenticated:
+        return HttpResponseRedirect("/account/login/")
+
+    utc_time = datetime.utcnow()
+    
+    selected_category = cmod.Category.objects.filter(id=category_id)
+    if selected_category.count() > 0:
+        prod_count = int(cmod.Product.objects.filter(category = category_id).count())
+        page_title = selected_category[0].name
+        print(page_title)
+        print("found selected category")
+    else:
+        prod_count = int(cmod.Product.objects.all().count())
+        page_title = 'Products'
+        print("didn't find category")
+
+
+    num_pages = 0
+    if prod_count % 6 > 0:
+        num_pages += 1
+    num_pages += (prod_count / 6)
+    num_pages = int(num_pages)
+
+    start_num = int(0)
```

Comment:
I'm not sure it's necessary to cast a hardcoded constant 0 to an int, though if you have to do this this is good to know.

Reply:
I don't think you have to either, but I was running into some issues so I went ahead and cast everything to be safe